RecoPixelVertexing/PixelTrackFitting/src/PixelTrackCleanerBySharedHits.cc

```diff
@@ -25,49 +27,89 @@ void PixelTrackCleanerBySharedHits::cleanTracks(TracksWithTTRHs & trackHitPairs)
   unsigned int size = trackHitPairs.size();
   if (size <= 1) return;
 
-  auto kill = [&](unsigned int i) { delete trackHitPairs[i].first; trackHitPairs[i].first=nullptr;};
+  // sort (stabilize cleaning)
+  uint16_t score[size]; 
+  unsigned int ind[size];
+  for (auto i = 0U; i < size; ++i) {
+     ind[i]=i; 
+     score[i]= 32000-std::min(32000, int(trackHitPairs[i].first->chi2()*100.f));  // chi2: smaller is better
+     if (trackHitPairs[i].second.size()==4) score[i]+=32001; //  s4 always better than s3
+  }
+  std::sort(ind,ind+size,[&](unsigned int i, unsigned int j){return score[i]>score[j];});
+
+  auto kill = [&](unsigned int k) {delete trackHitPairs[k].first; trackHitPairs[k].first=nullptr;};
```

Comment:
why change the argument from `i` to `k` ?

Reply:
no reason